lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -127,15 +127,19 @@ defmodule Mix.Compilers.Elixir do
 
     {compile, runtime} = Kernel.LexicalTracker.remotes(module)
```

Comment:
Yeah, we could do the pre-processing inside LexicalTracker.remotes itself and return a uniq list of module names.